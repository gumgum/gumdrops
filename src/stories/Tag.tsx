import React, { MouseEventHandler, ReactChild, ReactChildren } from 'react';
import cx from 'classnames';
import { Colors } from 'types';

export interface TagProps extends React.HTMLAttributes<Element> {
    color?: Colors;
    size?: 'xs' | 'sm';
    onOptionClick?: MouseEventHandler;
    hasOption?: boolean;
    optionIcon?: string;
    optionLabel?: string;
    text: ReactChild | ReactChildren | Element;
}

export const Tag: React.FC<TagProps> = ({
    color,
    size = 'sm',
    className,
    text,
    onOptionClick,
    hasOption = false,
    optionIcon = 'bt-times',
    optionLabel = 'Remove tag',
    ...otherProps
}) => {
    const rootClass = cx('gds-tag', className, {
        [`gds-tag--${color}`]: color,
        'gds-tag--with-button': hasOption,
        [`gds-tag--with-button--${size}`]: size && hasOption,
        [`gds-tag--${size}`]: size
    });

    const optionClass = cx('gds-tag__option', {
        [`gds-tag__option--${color}`]: color && hasOption,
        [`gds-tag__option--${size}`]: size
    });

    const iconClass = cx('btl', 'bt-fw', optionIcon);

    return (
        <div className={rootClass} {...otherProps}>
            {text}
            {hasOption && (
                <button aria-label={optionLabel} className={optionClass} onClick={onOptionClick}>
                    <i className={iconClass} />
                </button>
            )}
        </div>
    );
};
