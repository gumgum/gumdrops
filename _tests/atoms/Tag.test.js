import renderer from 'react-test-renderer';
import React from 'react';
import Tag from '../../components/atoms/Tag';

const defaultProps = {
    context: 'primary',
    className: 'custom-class',
    onClick: jest.fn(),
    onOptionClick: jest.fn(),
    hasOption: true,
    optionIcon: 'bt-times',
    optionLabel: 'Remove me',
    size: 'sm',
    style: { wdith: '100%' },
    text: 'Cool Tagz'
};

test('Expect <Tag> to render properly', () => {
    const tree = renderer.create(<Tag {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Expect <Button> to render properly without option', () => {
    const props = {
        ...defaultProps,
        hasOption: false,
        onOptionClick: undefined,
        optionIcon: undefined
    };
    const tree = renderer.create(<Tag {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
});
