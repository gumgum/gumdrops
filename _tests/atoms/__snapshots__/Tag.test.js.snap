// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Button> to render properly without option 1`] = `
<div
  className="gds-tag custom-class gds-tag--primary gds-tag--sm"
  onClick={[MockFunction]}
  style={
    {
      "wdith": "100%",
    }
  }
>
  Cool Tagz
</div>
`;

exports[`Expect <Tag> to render properly 1`] = `
<div
  className="gds-tag custom-class gds-tag--primary gds-tag--with-button gds-tag--with-button--sm gds-tag--sm"
  onClick={[MockFunction]}
  style={
    {
      "wdith": "100%",
    }
  }
>
  Cool Tagz
  <button
    aria-label="Remove me"
    className="gds-tag__option gds-tag__option--primary gds-tag__option--sm"
    onClick={[MockFunction]}
  >
    <i
      className="btl bt-fw bt-times"
    />
  </button>
</div>
`;
